pkg/kubelet/secret_manager.go

```diff
@@ -39,3 +54,181 @@ func newSimpleSecretManager(kubeClient clientset.Interface) (secretManager, erro
 func (s *simpleSecretManager) GetSecret(namespace, name string) (*api.Secret, error) {
 	return s.kubeClient.Core().Secrets(namespace).Get(name)
 }
+
+func (s *simpleSecretManager) RegisterPod(pod *api.Pod) {
+}
+
+func (s *simpleSecretManager) UnregisterPod(pod *api.Pod) {
+}
+
+// secretStoreItems is a single item stored in secretStore.
+type secretStoreItem struct {
+	secret   *api.Secret
+	refCount int
+}
+
+// secretStore is a local cache of secrets.
+type secretStore struct {
+	kubeClient clientset.Interface
+
+	sync.Mutex
+	items map[string]*secretStoreItem
+}
+
+func newSecretStore(kubeClient clientset.Interface) *secretStore {
+	ss := &secretStore{
+		kubeClient: kubeClient,
+		items:      make(map[string]*secretStoreItem),
+	}
+	go wait.NonSlidingUntil(func() { ss.Refresh() }, time.Minute, wait.NeverStop)
+	return ss
+}
+
+func keyFunc(namespace, name string) string {
+	return namespace + "/" + name
+}
+
+func namespaceNameFunc(key string) (string, string, error) {
+	items := strings.Split(key, "/")
+	if len(items) != 2 {
+		return "", "", fmt.Errorf("invalid key: %s", key)
+	}
+	return items[0], items[1], nil
+}
+
+func (s *secretStore) Add(namespace, name string) {
+	key := keyFunc(namespace, name)
+	secret, err := s.kubeClient.Core().Secrets(namespace).Get(name)
+	if err != nil {
+		glog.Errorf("Unable to retrieve a secret %s/%s: %v", namespace, name, err)
```

Comment:
It's what I wanted. Basically we have a pod that is referencing a secret "namespace/name" but we weren't able to fetch it. But hopefully when we will be refreshing it, we will fetch it and then update a pod.